pkg/credentials/config.go

```diff
@@ -8,22 +8,34 @@ import (
 	"net/url"
 	"os"
 
+	"gopkg.in/yaml.v3"
 	"k8s.io/client-go/rest"
 	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
 	clientcmdlatest "k8s.io/client-go/tools/clientcmd/api/latest"
 	"k8s.io/klog/v2"
 	"sigs.k8s.io/cluster-inventory-api/apis/v1alpha1"
 )
 
-// client.authentication.k8s.io/exec is a reserved extension key defined by the Kubernetes
-// client authentication API (SIG Auth), not by the ClusterProfile API.
-// Reference:
-// https://kubernetes.io/docs/reference/config-api/client-authentication.v1beta1/#client-authentication-k8s-io-v1beta1-Cluster
-const clusterExtensionKey = "client.authentication.k8s.io/exec"
+const (
+	// client.authentication.k8s.io/exec is a reserved extension key defined by the Kubernetes
+	// client authentication API (SIG Auth), not by the ClusterProfile API.
+	// Reference:
+	// https://kubernetes.io/docs/reference/config-api/client-authentication.v1beta1/#client-authentication-k8s-io-v1beta1-Cluster
+	clusterExecExtensionKey = "client.authentication.k8s.io/exec"
+
+	// additionalCLIArgsExtensionKey and additionalEnvVarsExtensionKey are
+	// two reserved extensions defined in KEP 5339, which allows users to pass in (usually cluster-specific)
+	// additional command-line arguments and environment variables to the exec plugin from
+	// the ClusterProfile API side.
+	additionalCLIArgsExtensionKey = "multicluster.x-k8s.io/clusterprofiles/auth/exec/additional-args"
+	additionalEnvVarsExtensionKey = "multicluster.x-k8s.io/clusterprofiles/auth/exec/additional-envs"
+)
 
 type Provider struct {
-	Name       string                   `json:"name"`
-	ExecConfig *clientcmdapi.ExecConfig `json:"execConfig"`
+	Name                       string                   `json:"name"`
+	ExecConfig                 *clientcmdapi.ExecConfig `json:"execConfig"`
+	AllowProfileSourcedCLIArgs bool                     `json:"allowProfileSourcedCLIArgs,omitempty"`
```

Comment:
Instead of boolean, how do you think if we can set a this as `Merge`, `Replace`, or `Ignore` ?